contracts/sfc/SFC.sol

```diff
@@ -901,6 +920,70 @@ contract SFC is Initializable, Ownable, Version {
         }
     }
 
+    /// Seal epoch - recalculate average uptime time of validators
+    function _sealEpochAverageUptime(
+        uint256 epochDuration,
+        EpochSnapshot storage snapshot,
+        EpochSnapshot storage prevSnapshot,
+        uint256[] memory validatorIDs,
+        uint256[] memory uptimes
+    ) internal {
+        for (uint256 i = 0; i < validatorIDs.length; i++) {
+            uint256 validatorID = validatorIDs[i];
+            // compute normalised uptime as a percentage in the Q1.30 fixed-point format
+            uint256 normalisedUptime = (uptimes[i] * (1 << 30)) / epochDuration;
```

Comment:
1 << 30 =>  1 << 31 (Perhaps use a constant somewhere for the fixpoint such as `const UQ1.0 = 1 << 31;` and use the symbol instead of 1<<31.